CONFIGURATION.md

```diff
@@ -59,12 +59,20 @@ ssl.cipher.suites                        |  *  |                 |
 ssl.curves.list                          |  *  |                 |               | low        | The supported-curves extension in the TLS ClientHello message specifies the curves (standard/named, or 'explicit' GF(2^k) or GF(p)) the client is willing to have the server use. See manual page for `SSL_CTX_set1_curves_list(3)`. OpenSSL >= 1.0.2 required. <br>*Type: string*
 ssl.sigalgs.list                         |  *  |                 |               | low        | The client uses the TLS ClientHello signature_algorithms extension to indicate to the server which signature/hash algorithm pairs may be used in digital signatures. See manual page for `SSL_CTX_set1_sigalgs_list(3)`. OpenSSL >= 1.0.2 required. <br>*Type: string*
 ssl.key.location                         |  *  |                 |               | low        | Path to client's private key (PEM) used for authentication. <br>*Type: string*
-ssl.key.password                         |  *  |                 |               | low        | Private key passphrase <br>*Type: string*
+ssl.key.password                         |  *  |                 |               | low        | Private key passphrase (for use with ssl.key.location) <br>*Type: string*
+ssl.key.pem                              |  *  |                 |               | low        | Client's private key string (PEM format) used for authentication. <br>*Type: string*
+ssl_key                                  |  *  |                 |               | low        | Client's private key as set by rd_kafka_conf_set_ssl_cert() <br>*Type: *
 ssl.certificate.location                 |  *  |                 |               | low        | Path to client's public key (PEM) used for authentication. <br>*Type: string*
-ssl.ca.location                          |  *  |                 |               | medium     | File or directory path to CA certificate(s) for verifying the broker's key. <br>*Type: string*
+ssl.certificate.pem                      |  *  |                 |               | low        | Client's public key string (PEM format) used for authentication. <br>*Type: string*
+ssl_certificate                          |  *  |                 |               | low        | Client's public key as set by rd_kafka_conf_set_ssl_cert() <br>*Type: *
+ssl.ca.location                          |  *  |                 |               | low        | File or directory path to CA certificate(s) for verifying the broker's key. <br>*Type: string*
+ssl_ca                                   |  *  |                 |               | low        | CA certificate as set by rd_kafka_conf_set_ssl_cert() <br>*Type: *
```

Comment:
I guess not having a string config property for CA certs was deliberate? Seems to be the only thing missing to complete the set. I don't know why it would be useful though.

Reply:
Actually, there are several new settings that offer in-memory certificates; hence the title for this PR. I believe for example that the string config property for CA certs is ssl_ca.

Reply:
@mhowlett Right, I don't believe it is useful. We can add it later if necessary.

@wesnerm ssl_ca is the backing store for set_ssl_cert(), it can't be set directly with a string setter.